storagedriver/factory/factory.go

```diff
@@ -58,7 +58,12 @@ func Create(name string, parameters map[string]interface{}) (storagedriver.Stora
 		// }
 		// return driverClient, nil
 	}
-	return driverFactory.Create(parameters)
+	d, err := driverFactory.Create(parameters)
```

Comment:
The driver factories should be modified to generate correct implementations. This can probably be done in `factory.Register`. It should probably also check that the driver has not already been wrapped.